src/main/java/bookbob/functions/CommandHandler.java

```diff
@@ -792,6 +799,101 @@ public void removePastAppointments(AppointmentRecord appointmentRecord) throws I
+            int nricStart = input.indexOf("ic/");
+            if (nricStart == -1) {
+                System.out.println("Please provide the patient's NRIC.");
+                return;
+            }
+            int nricEnd = findNextFieldStart(input, nricStart + 3);
+            String nric = input.substring(nricStart + 3, nricEnd).trim();
+
+            // Find the patient with matching NRIC
+            Patient targetPatient = null;
+            for (Patient patient : records.getPatients()) {
+                if (patient.getNric().equals(nric)) {
+                    targetPatient = patient;
+                    break;
+                }
+            }
+
+            if (targetPatient == null) {
+                System.out.println("No patient found with NRIC: " + nric);
+                return;
+            }
+
+            // Extract visit date (mandatory field)
+            int visitStart = input.indexOf("v/");
+            if (visitStart == -1) {
+                System.out.println("Please provide the visit date and time.");
+                return;
+            }
+            int visitEnd = findNextFieldStart(input, visitStart + 2);
+            String visitDateString = input.substring(visitStart + 2, visitEnd).trim();
+
+            // Parse visit date
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm");
+            LocalDateTime visitDate;
+            try {
+                visitDate = LocalDateTime.parse(visitDateString, formatter);
+            } catch (DateTimeParseException e) {
+                System.out.println("Invalid date format. Please use dd-MM-yyyy HH:mm format (e.g., 21-10-2024 15:48)");
+                return;
+            }
+
+            // Extract medications (optional, can be multiple)
+            ArrayList<String> medications = new ArrayList<>();
+            int medicationStart = input.indexOf("m/");
+            if (medicationStart != -1) {
+                int medicationEnd = findNextFieldStart(input, medicationStart + 2);
+                String medicationInput = input.substring(medicationStart + 2, medicationEnd).trim();
+                if (!medicationInput.isEmpty()) {
+                    String[] medsArray = medicationInput.split(",\\s*");
+                    medications.addAll(Arrays.asList(medsArray));
+                }
+            }
+
+            // Extract diagnoses (optional, can be multiple)
+            ArrayList<String> diagnoses = new ArrayList<>();
+            int diagnosisStart = input.indexOf("d/");
+            if (diagnosisStart != -1) {
+                int diagnosisEnd = findNextFieldStart(input, diagnosisStart + 2);
+                String diagnosisInput = input.substring(diagnosisStart + 2, diagnosisEnd).trim();
+                if (!diagnosisInput.isEmpty()) {
+                    String[] diagnosisArray = diagnosisInput.split(",\\s*");
+                    diagnoses.addAll(Arrays.asList(diagnosisArray));
+                }
+            }
```

Comment:
Perhaps can consider avoiding deeply nested codes (arrowhead codes)

Reply:
actually just thought of it, your suggestion of guard clauses is really good. if have time, will try refactor the code